base/bitarray.jl

```diff
@@ -1019,6 +1021,13 @@ function deleteat!(B::BitVector, inds)
     return B
 end
 
+function deleteat!(B::BitVector, inds::AbstractVector{Bool})
+    if length(B) != length(inds)
+        throw(ArgumentError("length of Bool indices inds must match the length of B"))
+    end
+    return deleteat!(B, findall(inds))
```

Comment:
clearly this can be made faster, but will lead to some code duplication. Should I improve things here?

Reply:
Seems fine for now. Would it be faster to reuse the code for Vector (at least avoiding the allocation)? The Vector method should work if `_deleteend!` is just replaced with `resize!`.

Reply:
no - it would not be fully efficient. I will implement an efficient method and you will judge.